### PR TITLE
fix: types in admin-settings package

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Fix types for `admin-settings` package.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue [hunk 1]
```diff
@@ -179,9 +189,10 @@ export default defineComponent({
       })
     })
 
-    const orderBy = (list, prop, desc) => {
+    const orderBy = (list: SpaceResource[], prop: string, desc: boolean) => {
+      console.log(prop, list)
```
Comment (dschmidt) [inline packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue:193]: 👀

File: packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue [hunk 2]
```diff
@@ -179,9 +189,10 @@ export default defineComponent({
       })
     })
 
-    const orderBy = (list, prop, desc) => {
+    const orderBy = (list: SpaceResource[], prop: string, desc: boolean) => {
```
Comment (dschmidt) [inline packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue:192]: same here?

File: packages/web-app-admin-settings/src/components/Users/UsersList.vue [hunk 3]
```diff
@@ -249,9 +256,9 @@ export default defineComponent({
       )
     }
 
-    const orderBy = (list, prop, desc) => {
+    const orderBy = (list: User[], prop: string, desc: boolean) => {
```
Comment (dschmidt) [inline packages/web-app-admin-settings/src/components/Users/UsersList.vue:259]: and here?

File: packages/web-app-admin-settings/src/index.ts [hunk 4]
```diff
@@ -13,13 +13,13 @@ import {
 import { RouteRecordRaw } from 'vue-router'
 
 // just a dummy function to trick gettext tools
-function $gettext(msg) {
+function $gettext(msg: string) {
   return msg
 }
```
Comment (dschmidt) [inline packages/web-app-admin-settings/src/index.ts:18]: can't we get rid of this function instead? 

we could do that in a followup as well (and get rid of all of them)...

File: packages/web-app-admin-settings/src/views/Spaces.vue [hunk 5]
```diff
@@ -120,7 +121,7 @@ export default defineComponent({
         'name asc',
         'driveType eq project'
       )
-      const drives = drivesResponse.map((space) =>
+      const drives = drivesResponse.map((space: Drive) =>
```
Comment (dschmidt) [inline packages/web-app-admin-settings/src/views/Spaces.vue:124]: why does this need an explicit type? If drivesResponse is properly typed, this should not be necessary

File: packages/web-app-admin-settings/src/views/Users.vue [hunk 6]
```diff
@@ -250,7 +252,7 @@ export default defineComponent({
       }
 
       const filter = Object.values(filters)
-        .reduce((acc, f: any) => {
```
Comment (dschmidt) [inline packages/web-app-admin-settings/src/views/Users.vue:253]: Nice :D

File: packages/web-app-admin-settings/tests/unit/components/Spaces/SideBar/ActionsPanel.spec.ts [hunk 7]
```diff
@@ -11,7 +11,7 @@ import {
 import { computed, ref } from 'vue'
 import { Action } from '@ownclouders/web-pkg'
 
-function createMockActionComposables(module) {
+function createMockActionComposables(module: Record<string, any>) {
```
Comment (dschmidt) [inline packages/web-app-admin-settings/tests/unit/components/Spaces/SideBar/ActionsPanel.spec.ts:14]: can we use Record<string, unknown>?

File: packages/web-app-admin-settings/tests/unit/components/Spaces/SpacesList.spec.ts [hunk 8]
```diff
@@ -69,7 +69,7 @@ describe('SpacesList', () => {
     'sorts by property "%s"',
     async (prop) => {
       const { wrapper } = getWrapper({ mountType: shallowMount, spaces: spaceMocks })
-      wrapper.vm.sortBy = prop
+      wrapper.vm.sortBy = prop as keyof SpaceResource
```
Comment (dschmidt) [inline packages/web-app-admin-settings/tests/unit/components/Spaces/SpacesList.spec.ts:72]: hmm ... can't we do this better here?

File: packages/web-app-admin-settings/tests/unit/index.spec.ts [hunk 9]
```diff
@@ -48,44 +48,32 @@ describe('admin settings index', () => {
       it('should redirect to general if permission given', () => {
         const ability = mock<Ability>()
         ability.can.mockReturnValueOnce(true)
-        expect(
-          routes({ $ability: ability })
-            .find((n) => n.path === '/')
-            .redirect().name
-        ).toEqual('admin-settings-general')
+        const route = routes({ $ability: ability }).find((n) => n.path === '/')
+        expect((route.redirect as any)().name).toEqual('admin-settings-general')
```
Comment (dschmidt) [inline packages/web-app-admin-settings/tests/unit/index.spec.ts:52]: 👀 

Vue Router Types can be such a pain to deal with...

File: packages/web-pkg/src/components/FilesList/ResourceTable.vue [hunk 10]
```diff
@@ -910,7 +919,11 @@ export default defineComponent({
       if (!this.isResourceSelected(item)) {
         this.emitSelect([item.id])
       }
-      displayPositionedDropdown(dropdown.tippy, event, this.$refs.contextMenuButton)
+      displayPositionedDropdown(
+        dropdown.tippy,
+        event,
+        this.$refs.contextMenuButton as ComponentPublicInstance<typeof OcButton>
```
Comment (dschmidt) [inline packages/web-pkg/src/components/FilesList/ResourceTable.vue:925]: hrm - why isn't this better typed and needs the cast? :-\

File: packages/web-pkg/src/helpers/contextMenuDropdown.ts [hunk 11]
```diff
@@ -1,8 +1,21 @@
...
+export const displayPositionedDropdown = (
+  dropdown: ComponentPublicInstance<typeof OcDrop>,
+  event: MouseEvent,
+  contextMenuButton: ComponentPublicInstance<typeof OcButton>
+) => {
   const contextMenuButtonPos = contextMenuButton.$el.getBoundingClientRect()
   const isKeyboardEvent = event.clientY === 0
   const yValue = isKeyboardEvent
-    ? event.srcElement?.getBoundingClientRect().top || 0
+    ? (event.target as HTMLElement)?.getBoundingClientRect().top || 0
```
Comment (dschmidt) [inline packages/web-pkg/src/helpers/contextMenuDropdown.ts:18]: huh - was this semantic change on purpose?

File: packages/web-pkg/src/services/eventBus.ts [hunk 12]
```diff
@@ -7,10 +7,10 @@ export class EventBus {
     this.topics = new Map()
   }
 
-  public subscribe(topic: string, callback: (data?: unknown) => void): string {
+  public subscribe<T = unknown>(topic: string, callback: (data?: T) => void): string {
     const subscription = {
       token: uuid.v4(),
-      callback
+      callback: callback as (data?: unknown) => void
```
Comment (dschmidt) [inline packages/web-pkg/src/services/eventBus.ts:13]: why can't the argument be typed better?

**Review by dschmidt — APPROVED** [review]

File: packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue (hunk 2 above)
Comment (JammingBen) [reply packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue:192]: As discussed, not possible here.

File: packages/web-app-admin-settings/src/components/Users/UsersList.vue (hunk 3 above)
Comment (JammingBen) [reply packages/web-app-admin-settings/src/components/Users/UsersList.vue:259]: As discussed, not possible here.

File: packages/web-app-admin-settings/src/index.ts (hunk 4 above)
Comment (JammingBen) [reply packages/web-app-admin-settings/src/index.ts:18]: As discussed, let's do this in a follow-up

File: packages/web-pkg/src/helpers/contextMenuDropdown.ts (hunk 11 above)
Comment (JammingBen) [reply packages/web-pkg/src/helpers/contextMenuDropdown.ts:22]: As discussed, `srcElement` is deprecated in favour of `target`.

File: packages/web-pkg/src/services/eventBus.ts (hunk 12 above)
Comment (JammingBen) [reply packages/web-pkg/src/services/eventBus.ts:13]: As discussed, currently the most convenient approach this way.

**Pull request merged**